Announcements

itbel opened the pull request:
- AnnouncementCard adjustments
- AnnouncementService no longer hardcoded
- AnnouncementDetailsScreen with and without pictures

resolves #158 
┆Issue is synchronized with this [Wrike Task](https://www.wrike.com/open.htm?id=618970767)

itbel committed:
moved query to queries.ts

itbel committed:
added announcementDetailscreen and card changes,

itbel committed:
homescreen changes

JonEsparaz commented on src/components/home/AnnouncementCard.tsx:
```diff
@@ -28,20 +39,59 @@ const style = StyleSheet.create({
+          />
+          <Text style={style.title}>{announcement.title}</Text>
+          <Text style={style.body}>
+            {`${announcement.description.slice(0, 100).trim()}...`}
```
React Native has props to handle this behaviour:
https://reactnative.dev/docs/text#numberoflines
https://reactnative.dev/docs/text#ellipsizemode

JonEsparaz commented on src/components/home/AnnouncementCard.tsx:
```diff
@@ -28,20 +39,59 @@ const style = StyleSheet.create({
-      <Text style={style.title}>{announcement.title}</Text>
-      <Text style={style.body}>{announcement.description}</Text>
+      <Text style={style.title}>{announcement?.title}</Text>
+      <Text style={style.body}>
```
Same as above - use numberOfLines and ellipsizeMode

JonEsparaz commented on src/screens/home/AnnouncementDetailsScreen.tsx:
```diff
@@ -60,33 +62,85 @@ export default function AnnouncementDetailScreen({
+            style={{ height: 56, marginBottom: 40, marginTop: 40 }}
+            label="Call to Action"
+            onPress={() => {
+              Linking.openURL(parseUrl() ?? '');
```
Is the nullish coalescing operator needed? Looks `parseUrl()` has default return value `''`

JonEsparaz commented on src/services/AnnouncementService.ts:
```diff
@@ -1,24 +1,54 @@
+    const today = moment()
+      .utcOffset(moment().isDST() ? '-0400' : '-0500')
+      .format('YYYY-MM-DD');
+    const queryResult = await runGraphQLQuery({
```
IMO we should not continue using `runGraphQLQuery()` (we should work towards removing it from the codebase).

Perhaps use the AWS Amplify SDK functions.

JonEsparaz commented on src/services/AnnouncementService.ts:
```diff
@@ -1,24 +1,54 @@
+import LocationService, { Location } from './LocationsService';
+import { listAnnouncements } from './queries';
+
+export type Announcement = {
```
This type can probably come from API.ts

itbel committed:
adjusted font sizes
now using ellipsis and numberoflines

itbel committed:
updated API.ts
announcemnts now use type from API.ts
now using graphqlOperation instead of runGraphQLQuery
removed unnecessary nullish coalescing from AnnouncementDetailsScreen.tsx
